tests/unit/test_cte.py

```diff
@@ -84,7 +84,7 @@ def test_find_duplicate_subtrees(test_case):
     assert repeated_node_complexity == expected_repeated_node_complexity
 
 
-def test_encode_node_id_with_query_select_sql(mock_analyzer):
+def test_encode_node_id_with_query_select_sql(mock_session, mock_analyzer):
```

Comment:
is mock_session used in this test?

Reply:
same as jamison's question. Linking old response: https://github.com/snowflakedb/snowpark-python/pull/2912#discussion_r1927517894